src/Viewer/components/MenuBar/MenuBar.js

```diff
@@ -167,6 +188,111 @@ export function MenuBar ({
+        setDownloadingMessage("Decoding data to database...");
+        setIsDownloading(true);
+        if (downloadWorker.current) {
+            downloadWorker.current.terminate();
+        }
+
+        const worker = new Worker(new URL("./downloadWorker.js", import.meta.url));
+        downloadWorker.current = worker;
+
+        worker.postMessage({
+            code: DOWNLOAD_WORKER_ACTION.initialize,
+            name: fileMetaData.name,
+            count: logFileState.downloadPageChunks,
+        });
+
+        let page = 1;
+        const blob = new BlobAppender();
+        worker.onmessage = (e) => {
+            const msg = e.data;
+            switch (msg.code) {
+                case DOWNLOAD_WORKER_ACTION.pageData:
+                    blob.append(msg.data + "\n");
+                    console.debug(`Added page ${msg.page} to stream.`);
+                    if (page <= logFileState.downloadPageChunks) {
+                        setProgress(90 + ((page/logFileState.downloadPageChunks) * 10));
+                        worker.postMessage({
+                            code: DOWNLOAD_WORKER_ACTION.pageData,
+                            page: page++,
+                        });
+                    } else {
+                        setIsDownloading(false);
+                        worker.postMessage({
+                            code: DOWNLOAD_WORKER_ACTION.clearDatabase,
+                        });
+                        worker.terminate();
+                        downloadBlob(blob.getBlob(), fileMetaData.name);
+                    }
+                    break;
+                case DOWNLOAD_WORKER_ACTION.progress:
+                    setProgress(msg.progress);
+                    if (msg.done) {
+                        setDownloadingMessage("Adding logs to stream...");
+                        setProgress(0);
+                        worker.postMessage({
+                            code: DOWNLOAD_WORKER_ACTION.pageData,
+                            page: page++,
+                        });
+                    }
+                    break;
+                case DOWNLOAD_WORKER_ACTION.clearDatabase:
+                    setIsDownloading(false);
+                    worker.terminate();
+                    console.error(msg.error);
+                    break;
+                case DOWNLOAD_WORKER_ACTION.error:
+                    console.error(msg.error);
+                    break;
+                default:
+                    break;
+            }
+        };
+        worker.postMessage({
+            code: DOWNLOAD_WORKER_ACTION.progress,
+        });
+    };
+
+    const hasFilePath = () => {
+        const urlParams = new URLSearchParams(window.location.search);
+        return urlParams.get("filePath");
+    };
+
+    window.onbeforeunload = function (e) {
+        changeStateCallback(STATE_CHANGE_TYPE.stopDownload, null);
+        if (downloadWorker.current !== null) {
```

Comment:
```suggestion
        if (null !== downloadWorker.current) {
```
We should use Yoda condition: https://en.wikipedia.org/wiki/Yoda_conditions